src/security/tests/acl_api_tests.c

```diff
@@ -1406,7 +1898,35 @@ main(void)
 		cmocka_unit_test(test_acl_remove_ace_first),
 		cmocka_unit_test(test_acl_remove_ace_last),
 		cmocka_unit_test(test_acl_remove_ace_with_name),
-		cmocka_unit_test(test_acl_remove_ace_not_found)
+		cmocka_unit_test(test_acl_remove_ace_not_found),
+		cmocka_unit_test(test_ace_is_valid_null),
+		cmocka_unit_test(test_ace_is_valid_valid_types),
+		cmocka_unit_test(test_ace_is_valid_invalid_owner),
+		cmocka_unit_test(test_ace_is_valid_invalid_user),
+		cmocka_unit_test(test_ace_is_valid_invalid_owner_group),
+		cmocka_unit_test(test_ace_is_valid_invalid_group),
+		cmocka_unit_test(test_ace_is_valid_invalid_everyone),
+		cmocka_unit_test(test_ace_is_valid_group_needs_flag),
+		cmocka_unit_test(test_ace_is_valid_non_group_needs_no_flag),
+		cmocka_unit_test(test_ace_is_valid_principal_len_not_aligned),
+		cmocka_unit_test(test_ace_is_valid_principal_not_terminated),
+		cmocka_unit_test(test_ace_is_valid_undefined_flags),
+		cmocka_unit_test(test_ace_is_valid_valid_flags),
+		cmocka_unit_test(test_ace_is_valid_undefined_perms),
+		cmocka_unit_test(test_ace_is_valid_valid_perms),
+		cmocka_unit_test(test_ace_is_valid_undefined_access_type),
+		cmocka_unit_test(test_ace_is_valid_valid_access_types),
+		cmocka_unit_test(test_ace_is_valid_perms_for_unset_type),
+		cmocka_unit_test(test_ace_is_valid_audit_flags_with_only_allow),
+		cmocka_unit_test(test_ace_is_valid_audit_without_flags),
+		cmocka_unit_test(test_acl_is_valid_null),
+		cmocka_unit_test(test_acl_is_valid_empty),
+		cmocka_unit_test(test_acl_is_valid_bad_version),
+		cmocka_unit_test(test_acl_is_valid_len_too_small),
+		cmocka_unit_test(test_acl_is_valid_len_unaligned),
+		cmocka_unit_test(test_acl_is_valid_one_invalid_ace),
+		cmocka_unit_test(test_acl_is_valid_valid_aces),
+		cmocka_unit_test(test_acl_is_valid_later_ace_invalid)
```

Comment:
Shall we add a test where we generate a random buffer and validate it?

Reply:
Good idea, I will add that.